src/test/ui/dollar-crate/dollar-crate-is-keyword-2.rs

```diff
@@ -2,7 +2,7 @@ mod a {}
 
 macro_rules! m {
     () => {
-        use a::$crate; //~ ERROR unresolved import `a::$crate`
+        use a::$crate; //~ ERROR unresolved import `a::crate`
```

Comment:
I am a bit surprised to see this, though I didn't try to trace the calls to the new infrastructure. I think it's a correct change though strange (to me) that how we print paths differs from how we print the errors below, I guess. But ultimately seems good -- it's a more correct error AFAICT.

Out of interest, in a cross-crate use, does the expansion break down and expand to `a::::crate_name` here?

Reply:
In one case it happens to be printed through `ident` (`ast::Ident`) and in another through `ident.name` (`ast::Name` aka `Symbol`).

This is all annoying.
Previously `format!("{}", ident)` and `format!("{}", ident.name)` were interchangeable, but in https://github.com/rust-lang/rust/pull/67010 @estebank wanted to print identifiers in error messages as raw if they are likely raw, so they are now different.

Now @olegnn in https://github.com/rust-lang/rust/pull/68963 wants to print `Symbol`s in the same way, with rawness guessing.
This, however needs to be opt-in because `Symbol` is not generally an identifier, raw or not, it's simply an interned string.

There are seems to be three cases for identifier printing with different requirements:
- Token printing. It is important for this one is precise, no rawness guessing, no `$crate` conversions. This is how proc macros know what tokens are passed to them.
This is also used for error messages in lexer/parser where we are dealing with tokens (through `fn token_to_string`). This is an excellent candidate for a `Display` impl for `Token`, which doesn't exist yet.
- AST printing. We need to print larger pieces of (possibly expanded) AST for both people, and for proc macros (due to [issues](https://github.com/rust-lang/rust/issues/43081)). Both people and proc macros are then trying to parse and further compile that pretty-printed code again, so this printing has to recover rawness (which is not stored in AST, but can be guessed very well) and convert `$crate`s, which simply don't parse otherwise.
- Printing individual identifiers in error messages. That's the use case that @estebank had in mind when implementing https://github.com/rust-lang/rust/pull/67010.
Apparently the rawness should be guessed as well in this case. Should the `$crate`s be converted though? I have no idea.
If the logic is that the printed error message should look as similar as possible to the source code, then it shouldn't.
I agree that this use case is better served by a `Display` impl (the most ergonomic way) than by an explicit function, because error reporting code is more often written by people than pretty-printing code which is already written once.

Reply:
I guess I'll change the PR to account for the "print `ast::Ident` in a way that is as similar as possible to its originally written token in source code" use case, which is important for error messages.

Reply:
And also move `token_to_string` to `Display for Token` while I'm here.